manual/settings.md

```diff
@@ -1253,6 +1253,16 @@ The maximum number of query warnings to log before muting them. Query warnings a
 
 Default: **50**.
 
+#### --log.printSources<br />--dsbulk.log.printSources _&lt;boolean&gt;_
+
+Whether to print record sources in debug files. When set to true (the default), debug files will contain, for each record that failed to be processed, its original source, such as the text line that the record was parsed from.
+
+This can be useful to locate failed records more easily and diagnose processing failures. Furthermore, when loading, enabling this option also enables the creation of so-called "bad files", that is, files containing the original lines that could not be inserted, which could then be used as the input file of a subsequent load operation to load only the failed records.
+
+But for this feature to be possible, record sources must be kept in memory until the record is fully processed. For large record sizes (over 1 megabyte per record), retaining record sources in memory could put a high pressure on the JVM heap, thus exposing the operation to out-of-memory errors. This phenomenon is exacerbated when batching is enabled. If you are experiencing such errors, consider disabling this option.
```

Comment:
+1 on making it default to "`true`".
I see one case that may be worth mentioning here: reading from external URL (HTTP, FTP). In such a case, setting `false` may be problematic because users may not be able to locate the original line.